code/game/machinery/deployable.dm

```diff
@@ -53,14 +56,19 @@
 		if(!anchored)
 			return TRUE
 		var/obj/item/projectile/proj = mover
+		if(directional_blockage)
+			if(turn(mover.dir, 180) in directional_list)
+				return FALSE
 		if(proj.firer && Adjacent(proj.firer))
 			return TRUE
 		if(prob(proj_pass_rate))
 			return TRUE
 		return FALSE
-	else
-		return !density
-
+	if(isitem(mover)) //thrown items with the dropwall
+		if(directional_blockage)
+			if(turn(mover.dir, 180) in directional_list)
+				return FALSE
```

Comment:
Prob best to make this into it's own proc

Reply:
Done